### PR TITLE
Ensure submodules are named consistently

### PR DESCRIPTION
**Pull request by rschu1ze**
Follow-up to https://github.com/ClickHouse/ClickHouse/pull/65048/ where the abseil submodule name had no "contrib/" prefix.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CI config
- [x]  Run only ASAN package
- [x]  and Style check

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 13b23e9e9c3465902ad57c6b1e802c2d54832e42 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65167/13b23e9e9c3465902ad57c6b1e802c2d54832e42/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Review by thevar1able — APPROVED** [review]

File: utils/check-style/check-style [hunk 1]
```diff
@@ -243,6 +243,10 @@ done
 # All the submodules should be from https://github.com/
 find $ROOT_PATH -name '.gitmodules' | while read i; do grep -F 'url = ' $i | grep -v -F 'https://github.com/' && echo 'All the submodules should be from https://github.com/'; done
 
+# All submodules should be of this form: [submodule "contrib/libxyz"] (for consistency, the submodule name does matter too much)
+# - restrict the check to top-level .gitmodules file
+find $ROOT_PATH -maxdepth 1 -name '.gitmodules' | while read i; do grep -F '[submodule ' $i | grep -v -F 'contrib' && echo 'All submodules should have form [submodule "contrib/libxyz"]'; done
```
Comment (Felixoid) [inline utils/check-style/check-style:248]:
```suggestion
git config --file .gitmodules --get-regexp 'submodule\..+\.path' | \
while read -r line; do
    name=${line#submodule.}; name=${name%.path*}
    path=${line#* }
    [ "$path" != "$path" ] && echo "Submodule name $name is not equal to it's path"
done
```

Let's use `git config` to get the values

Comment (Felixoid) [reply utils/check-style/check-style:248]: From the third or fourth time it works

```
$ git config --file "$ROOT_PATH/.gitmodules" --get-regexp 'submodule\..+\.path' | \
while read -r line; do
    name=${line#submodule.}; name=${name%.path*}
    path=${line#* }
    [ "$name" != "$path" ] && echo "Submodule name '$name' is not equal to it's path '$path'"
done
Submodule name 'aws-c-event-stream' is not equal to it's path 'contrib/aws-c-event-stream'
Submodule name 'aws-c-common' is not equal to it's path 'contrib/aws-c-common'
Submodule name 'aws-checksums' is not equal to it's path 'contrib/aws-checksums'
Submodule name 'abseil' is not equal to it's path 'contrib/abseil-cpp'
```

File: utils/check-style/check-style [hunk 2]
```diff
@@ -243,6 +243,15 @@ done
 # All the submodules should be from https://github.com/
 find $ROOT_PATH -name '.gitmodules' | while read i; do grep -F 'url = ' $i | grep -v -F 'https://github.com/' && echo 'All the submodules should be from https://github.com/'; done
```
Comment (Felixoid) [inline utils/check-style/check-style:244]:
```suggestion

git config --file "$ROOT_PATH/.gitmodules" --get-regexp 'submodule\..+\.url' | \
while read -r line; do 
    name=${line#submodule.}; name=${name%.url*}
    url=${line#* }
    [[ "$url" != 'https://github.com/'* ]] && echo "All the submodules should be from https://github.com/, submodule '$name' has '$url'"
done
```

Didn't see that there's another one above `¯\_(ツ)_/¯`

**Comment by Felixoid** [comment]
Hehe, it works https://s3.amazonaws.com/clickhouse-test-reports/65167/3fd1918c3f6f92aeb14119a461cde2ed4038cedd/style_check.html

Maybe even it was an accidental check =)

**Pull request merged**